### ISSUE TITLE
[3.3.1] Ironing in multi extruder setup irons too much

### ISSUE THREAD

**Issue by MB3Y** [Ultimaker/Cura#3743]
**R3.3.1**

**Win10**

**Rostock301 - diamond hotend with 3filaments through one nozzle**

[Fortnite_logo Flat Small.curaproject.3mf.zip](https://github.com/Ultimaker/Cura/files/1955924/Fortnite_logo.Flat.Small.curaproject.3mf.zip)

**Steps to Reproduce**
Use actively 3 extruders: outer wall, inner wall and skin.
Enable ironing.

**Actual Results**
The ironing action is executed 3 times in this example, i.e once per extruder. Nothing wrong with it - as long as each extruder would only iron it's own color, but each color irons the whole surface.

**Expected results**
I had expected that each color would only iron over itself.
Not only this is a waste of time and unnecessary that skin area is ironed once per extruder, but also the colors are not maintained :'(.

![2018-04-27](https://user-images.githubusercontent.com/29209302/39375807-e14a3476-4a4f-11e8-84b2-699036bbf73f.png)

**Comment by diegopradogesto** [comment]
Yes, I have checked and it's wrong behaviour. I will create an internal ticket for this and we will take a look at it as soon as we can.
For developers: CURA-5338

**Comment by diegopradogesto** [comment]
Our manager decided not to to this for now. He removed it from the planning.

**Comment by jeremyparadie** [comment]
I've run into this issue attempting to mass-produce many tiny parts with an Ultimaker S5. The extra time ironing the build volume once for each extruder adds up to be significant. Any updates or workarounds? I'm currently using Cura 4.6.2 and Cura 4.7.0-beta.

**Comment by Ghostkeeper** [comment]
Hmm, well I'm not able to reproduce it any more. It's ironing only the models printed with that extruder, for me:

![image](https://user-images.githubusercontent.com/2448634/90236265-cf83d800-de22-11ea-882b-61739287770b.png)

**Comment by jeremyparadie** [comment]
See the attached project file. It appears that both extruders take the same ironing path. Since it is the same path, it is hard to tell looking at a snapshot of the slice preview but if you play the simulation of the final layer or actually print it it should become more apparent that ironing occurs twice, once for each extruder. 

In the end, I decided that it looked better with two passes of ironing and since I couldn't do anything about it I ended up using this process. I had black PLA loaded in both extruders. so color mixing wasn't an issue. With that said, it would be ideal to have more control over which extruder is used for ironing and how many passes it should take. 

[single strut iron test for github.zip](https://github.com/Ultimaker/Cura/files/5178494/single.strut.iron.test.for.github.zip)

**Comment by Ghostkeeper** [comment]
Indeed, with that project file it reproduces. You can see it extruding the red material if you hide the second extruder in layer view.

**Comment by jeremyparadie** [comment]
Bad bot. I provided a project file that produces the issue.

**Comment by Ghostkeeper** [comment]
Sorry about that. The bot didn't register it because you're not the original author of the issue.

**Comment by ansonl** [comment]
@Ghostkeeper @diegopradogesto Has there been any update to fix this issue? I tried to slice a file with dual extrusion in Cura 4.13.1 with ironing and both extruders unnecessarily iron each shared layer. 

The issue goes away with ironing unchecked for both extruders but then that means no ironing exposed upper surfaces so the ironing feature is not very useful for dual extrusion prints. 

[UMODXUD_crystalcluster-2.zip](https://github.com/Ultimaker/Cura/files/9551054/UMODXUD_crystalcluster-2.zip)

![image](https://user-images.githubusercontent.com/546458/189737341-5d539e09-0358-43ad-9b6a-62574acc25a2.png)
![image](https://user-images.githubusercontent.com/546458/189737431-811bbb82-2c81-45a7-afd9-a05bbbfebafe.png)

**Comment by GregValiant** [comment]
See seems to have been fixed at some point.  I'll go ahead and close it.  Let me know of that is premature.
